MailCore/Models/MergedContact.swift

```diff
@@ -35,42 +41,47 @@ extension CNContact {
     }
 }
 
-public final class MergedContact {
+public final class MergedContact: Object, Identifiable {
     private static let contactFormatter = CNContactFormatter()
 
-    public var email: String
-    public var remote: Contact?
-    public var local: CNContact?
+    /// Shared
+    @Persisted(primaryKey: true) public var email: String
+    @Persisted public var name: String
 
-    public lazy var color: UIColor = {
-        if let remoteColorHex = remote?.color,
-           let colorFromHex = UIColor(hex: remoteColorHex) {
-            return colorFromHex
-        } else {
-            return UIColor.backgroundColor(from: email.hash, with: UIConstants.avatarColors)
-        }
-    }()
+    /// Remote
+    @Persisted public var remoteColorHex: String?
+    @Persisted public var remoteAvatarURL: String?
+    @Persisted public var remoteIdentifier: String?
 
-    public lazy var name: String = {
-        guard let local,
-              let localName = MergedContact.contactFormatter.string(from: local) else {
-            return remote?.name ?? ""
+    /// Local
+    @Persisted public var localIdentifier: String?
+
+    /// false if can point to a contact available on the iPhone contacts
+    public lazy var isRemote = {
+        guard let localIdentifier,
```

Comment:
@adrien-coye can you check the condition ? :)

Reply:
Nice catch. Done